sdk/cosmos/azure-spring-data-cosmos-core/src/main/java/com/azure/spring/data/cosmos/core/ReactiveCosmosTemplate.java

```diff
@@ -370,21 +351,37 @@ public void setApplicationContext(@NonNull ApplicationContext applicationContext
         generateIdIfNullAndAutoGenerationEnabled(objectToSave, domainType);
         final JsonNode originalItem = prepareToPersistAndConvertToItemProperties(objectToSave);
         final CosmosItemRequestOptions options = new CosmosItemRequestOptions();
+        CosmosAsyncContainer cosmosAsyncContainer = cosmosAsyncClient.getDatabase(this.databaseName)
+                                                                     .getContainer(containerName);
+        Mono<CosmosItemResponse<JsonNode>> item;
         if (partitionKey == null) {
-            partitionKey = PartitionKey.NONE;
+            //  if the partition key is null, SDK will get the partitionKey from the object
```

Comment:
Where is the code about `SDK will get the partitionKey from the object` ?

Reply:
Here is the logic : https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java#L1031